### PR TITLE
Run tests in CI using github actions

### PR DESCRIPTION
**Pull request by jhamon**
## Problem

This repository has some tests, but they are not being run in CI. We need to setup this integration so we can evaluate the impact of PRs and begin to collaborate with more people.

## Solution

Implement a github actions workflow to run tests.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Observe things running in Github Actions.

## Notes

- There's definitely some kind of test timing issue causing these tests to flake. But some tests is better than no tests.

### CODE REVIEW

File: .github/workflows/PR.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        run: | 
+          npm install --ignore-scripts
+      - name: Build typescript
+        run: |
+          npm run build
+      - name: Run tests
+        env:
+          CI: true
+          PINECONE_API_KEY: ${{ secrets.PINECONE_API_KEY }}
+          PINECONE_ENVIRONMENT: ${{ secrets.PINECONE_ENVIRONMENT }}
```
Comment (damargulis) [inline .github/workflows/PR.yml:36]: Which environment do you plan to run this in?

**Review by damargulis — APPROVED** [review]
Nice, looks great!

File: .github/workflows/PR.yml (hunk 1 above)
Comment (jhamon) [reply .github/workflows/PR.yml:36]: At the moment I just have them running in my personal free tier env, but need to sort out something else soon. Any suggestion?

Comment (jhamon) [reply .github/workflows/PR.yml:36]: We should probably discuss this somewhere less public.

**Review by rschwabco — APPROVED** [review]
Looks great!

File: tests/index.test.ts [hunk 2]
```diff
@@ -19,27 +23,34 @@ const quantity = 10
 const metric = 'dotproduct'
 const vectors = generateVectors(dimensions, quantity)
 const vectorsWithSparseValues = generateVectors(dimensions, quantity, true)
+const client: PineconeClient = new PineconeClient()
+
+beforeEach(() => {
+  jest.setTimeout(100000)
```
Comment (micmmakarov) [inline tests/index.test.ts:29]: Is this the maximum amount of time per all tests, 100 seconds?

**Review by micmmakarov — APPROVED** [review]

**Comment by miararoy** [comment]
LGTM, great stuff @jhamon 💪

File: tests/index.test.ts (hunk 2 above)
Comment (jhamon) [reply tests/index.test.ts:29]: I think it's the maximum time for an individual test. Some of the tests are doing time-consuming things like creating a new index, which involves waiting a bit before it is ready. In reality, most tests do not take anywhere near that long.

**Pull request merged**